### PR TITLE
Add UserContext and mock login/logout functionality

### LINKED ISSUE #56: Hard code user login functionality as stop gap
User will always be logged in as user1. Temporary measure for demo purposes.

### PR DESCRIPTION
**Pull request by leahriffell**
### What was changed?
- This is temporary functionality to simulate a logged in user (for demo purposes)
- When clicking login, you will always be logged in as user 1
- Will always display username for user 1 (API request for getUser hardcodes id)
- Adds React Router and UserContext

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.

### Tracking
- Closes #56 

### Tracking Consistency:
- [x] Added appropriate labels
- [x] Added comments on my pull request, particularly in hard-to-understand areas

### CODE REVIEW

File: cypress/integration/it_can_log_user_in_and_out.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  it('can logout user', () => {
+    cy.visit('/')
+
+    cy.contains('login').click()
+    cy.contains('logout').click()
+
+    cy.contains('Logged in as: LoganRools').should('not.exist')
+    cy.contains('login')
+  })
+})
```
Comment (leahriffell) [inline cypress/integration/it_can_log_user_in_and_out.js:20]: Just tests that username is displayed but doesn't actually test user context. Will add more robust test once this actual functionality is implemented.

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,22 +1,23 @@
-import React, { } from 'react';
-import Header from '../Header/Header';
```
Comment (leahriffell) [inline src/components/App/App.js:2]: Renamed Header component to Home

File: src/components/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  {
+    getUser(id: "1") {
+      username
+    }
+  }
+`;
+
+const Home = () => {
+  const { user, setUser } = useContext(UserContext);
+  const { loading, error, data } = useQuery(GET_USER);
```
Comment (leahriffell) [inline src/components/Home/Home.js:16]: Was hoping to only call this query IF there's a user but wasn't sure where this line could live. I tried adding it in the if user conditional but that caused errors living within a return. I was also hoping to pass user.id dynamically to the query but for now hardcoded on line 8.

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const login = async () => {
+  return {
+    id: 1
+  };
+};
```
Comment (leahriffell) [inline src/components/Login/Login.js:5]: Temporary stop gap (hard code return of user id 1)

File: cypress/integration/it_can_log_user_in_and_out.js (hunk 1 above)
Comment (damwhit) [reply cypress/integration/it_can_log_user_in_and_out.js:20]: This seems fine for now.

File: src/components/App/App.js [hunk 5]
```diff
@@ -1,22 +1,23 @@
...
+import React, { useState, useMemo } from 'react';
+import Home from '../Home/Home';
```
Comment (damwhit) [inline src/components/App/App.js:2]: Might consider naming file index to make import statement simpler.

**Review by damwhit — APPROVED** [review]
Seems fine for now. 🇹🇭

**Pull request merged**